libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.ts

```diff
@@ -292,14 +300,18 @@ export class UploadCollectionComponent {
     /**@hidden
      * Opens the file selector.
      */
-    _fileNameChange(e: FocusEvent, item: UploadCollectionItem): void {
+    _fileNameChange(e: FocusEvent, currentItem: UploadCollectionItem): void {
         const input = e.target as HTMLInputElement;
         const itemName = input.value;
-        const newName = item.type === 'file' ? `${itemName}.${item.name.split('.').pop()}` : itemName;
+        const newName = currentItem.type === 'file' ? `${itemName}.${currentItem.name.split('.').pop()}` : itemName;
+
+        if (currentItem.name === newName) {
+            return;
+        }
 
         if (itemName.length > this.maxFilenameLength) {
             const reanmedItem = {
```

Comment:
Please fix a typo 'reanmedItem` => `renamedItem`

Reply:
Typo still not fixed